include/secp256k1_schnorrsig.h

```diff
@@ -2,126 +2,108 @@
+    unsigned char *nonce32,
+    const unsigned char *msg32,
+    const unsigned char *key32,
+    const unsigned char *xonly_pk32,
+    const unsigned char *algo16,
+    void *data
+);
 
-/** Parse a Schnorr signature.
+/** An implementation of the nonce generation function as defined in Bitcoin
+ *  Improvement Proposal 340 "Schnorr Signatures for secp256k1"
+ *  (https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki).
  *
- *  Returns: 1 when the signature could be parsed, 0 otherwise.
- *  Args:    ctx: a secp256k1 context object
- *  Out:     sig: pointer to a signature object
- *  In:     in64: pointer to the 64-byte signature to be parsed
- *
- * The signature is serialized in the form R||s, where R is a 32-byte public
- * key (x-coordinate only; the y-coordinate is considered to be the unique
- * y-coordinate satisfying the curve equation that is a quadratic residue)
- * and s is a 32-byte big-endian scalar.
- *
- * After the call, sig will always be initialized. If parsing failed or the
- * encoded numbers are out of range, signature validation with it is
- * guaranteed to fail for every message and public key.
+ *  If a data pointer is passed, it is assumed to be a pointer to 32 bytes of
+ *  auxiliary random data as defined in BIP-340. If the data pointer is NULL,
+ *  schnorrsig_sign does not produce BIP-340 compliant signatures. The algo16
+ *  argument must be non-NULL, otherwise the function will fail and return 0.
+ *  The hash will be tagged with algo16 after removing all terminating null
+ *  bytes. Therefore, to create BIP-340 compliant signatures, algo16 must be set
+ *  to "BIP0340/nonce\0\0\0"
  */
-SECP256K1_API int secp256k1_schnorrsig_parse(
-    const secp256k1_context* ctx,
-    secp256k1_schnorrsig* sig,
-    const unsigned char *in64
-) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+SECP256K1_API extern const secp256k1_nonce_function_hardened secp256k1_nonce_function_bip340;
 
 /** Create a Schnorr signature.
  *
- * Returns 1 on success, 0 on failure.
+ *  Does _not_ strictly follow BIP-340 because it does not verify the resulting
+ *  signature. Instead, you can manually use secp256k1_schnorrsig_verify and
+ *  abort if it fails.
+ *
+ *  Otherwise BIP-340 compliant if the noncefp argument is NULL or
+ *  secp256k1_nonce_function_bip340 and the ndata argument is 32-byte auxiliary
+ *  randomness.
+ *
+ *  Returns 1 on success, 0 on failure.
  *  Args:    ctx: pointer to a context object, initialized for signing (cannot be NULL)
- *  Out:     sig: pointer to the returned signature (cannot be NULL)
- *       nonce_is_negated: a pointer to an integer indicates if signing algorithm negated the
- *                nonce (can be NULL)
- *  In:    msg32: the 32-byte message hash being signed (cannot be NULL)
- *        seckey: pointer to a 32-byte secret key (cannot be NULL)
- *       noncefp: pointer to a nonce generation function. If NULL, secp256k1_nonce_function_bipschnorr is used
```

Comment:
this nonce_function_bipschnorr still exists because it's in `secp256k1.c` and `secp256k1.h` and therefore it wasn't removed in the first commit (but it should)